modules/runes/api/httphandler/get_token_list.go

```diff
@@ -0,0 +1,171 @@
+package httphandler
+
+import (
+	"strings"
+
+	"github.com/cockroachdb/errors"
+	"github.com/gaze-network/indexer-network/common/errs"
+	"github.com/gaze-network/indexer-network/modules/runes/runes"
+	"github.com/gofiber/fiber/v2"
+	"github.com/samber/lo"
+)
+
+const (
+	getTokensMaxLimit = 1000
+)
+
+type GetTokensScope string
+
+const (
+	GetTokensScopeAll     GetTokensScope = "all"
+	GetTokensScopeMinting GetTokensScope = "minting"
+)
+
+func (s GetTokensScope) IsValid() bool {
+	switch s {
+	case GetTokensScopeAll, GetTokensScopeMinting:
+		return true
+	}
+	return false
+}
+
+type getTokensRequest struct {
+	paginationRequest
+	Search      string         `query:"search"`
+	BlockHeight uint64         `query:"blockHeight"`
+	Scope       GetTokensScope `query:"scope"`
+}
+
+func (req getTokensRequest) Validate() error {
+	var errList []error
+	if err := req.paginationRequest.Validate(); err != nil {
+		errList = append(errList, err)
+	}
+	if req.Limit > getTokensMaxLimit {
+		errList = append(errList, errors.Errorf("limit must be less than or equal to 1000"))
+	}
+	if req.Scope != "" && !req.Scope.IsValid() {
+		errList = append(errList, errors.Errorf("invalid scope: %s", req.Scope))
+	}
+	return errs.WithPublicMessage(errors.Join(errList...), "validation error")
+}
+
+func (req *getTokensRequest) ParseDefault() error {
+	if err := req.paginationRequest.ParseDefault(); err != nil {
+		return errors.WithStack(err)
+	}
+	if req.Scope == "" {
+		req.Scope = GetTokensScopeAll
+	}
+	return nil
+}
+
+type getTokensResult struct {
+	List []getTokenInfoResult `json:"list"`
+}
+
+type getTokensResponse = HttpResponse[getTokensResult]
+
+func (h *HttpHandler) GetTokens(ctx *fiber.Ctx) (err error) {
+	var req getTokensRequest
+	if err := ctx.QueryParser(&req); err != nil {
+		return errors.WithStack(err)
+	}
+	if err := req.Validate(); err != nil {
+		return errors.WithStack(err)
+	}
+	if err := req.ParseDefault(); err != nil {
+		return errors.WithStack(err)
+	}
+
+	blockHeight := req.BlockHeight
+	if blockHeight == 0 {
+		blockHeader, err := h.usecase.GetLatestBlock(ctx.UserContext())
+		if err != nil {
+			if errors.Is(err, errs.NotFound) {
+				return errs.NewPublicError("latest block not found")
+			}
+			return errors.Wrap(err, "error during GetLatestBlock")
+		}
+		blockHeight = uint64(blockHeader.Height)
+	}
+
+	// remove spacers
+	search := strings.Replace(strings.Replace(req.Search, "•", "", -1), ".", "", -1)
+
+	var entries []*runes.RuneEntry
+	switch req.Scope {
+	case GetTokensScopeAll:
+		entries, err = h.usecase.GetRuneEntries(ctx.UserContext(), search, blockHeight, req.Limit, req.Offset)
+		if err != nil {
+			return errors.Wrap(err, "error during GetRuneEntryList")
+		}
+	case GetTokensScopeMinting:
+		entries, err = h.usecase.GetMintingRuneEntries(ctx.UserContext(), search, blockHeight, req.Limit, req.Offset)
+		if err != nil {
+			return errors.Wrap(err, "error during GetRuneEntryList")
+		}
+	default:
+		panic("invalid scope")
```

Comment:
shouldn't panic in http handler, do return `errs.InternError` or something instead